sources/notion/__init__.py

```diff
@@ -27,15 +30,24 @@ def notion_databases(
     Yields:
         DltResource: Data resources from Notion databases.
     """
-    notion_client = NotionClient(api_key)
+    notion_client = APIClient(
+        base_url=API_URL,
+        headers=DEFAULT_HEADERS,
+        auth=BearerTokenAuth(api_key),
+        paginator=NotionPaginator(),
+    )
 
     if database_ids is None:
-        search_results = notion_client.search(
-            filter_criteria={"value": "database", "property": "object"}
+        search_results = notion_client.paginate(
```

Comment:
I guess also line 56 needs to be changed to use `paginate()` to get the databases